resources/assets/coffee/react/beatmap-discussions/user-filter.coffee

```diff
@@ -16,112 +16,65 @@
 #    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 ###
 
-{a, i, div} = ReactDOMFactories
+{a} = ReactDOMFactories
 el = React.createElement
 
-bn = 'beatmap-discussions-user-filter'
-
 allUsers =
   id: null,
-  username: osu.trans('beatmap_discussions.user_filter.everyone')
-
+  text: osu.trans('beatmap_discussions.user_filter.everyone')
 
 class BeatmapDiscussions.UserFilter extends React.PureComponent
-  constructor: (props) ->
-    super props
-
-    @state =
-      showingSelector: false
-
-
-  componentDidMount: =>
-    $(document).on 'click.userFilter', @hideSelector
-
-
-  componentDidUpdate: (_prevProps, prevState) =>
-    Blackout.toggle(@state.showingSelector, 0.5) unless prevState.showingSelector == @state.showingSelector
+  render: =>
+    options = [allUsers]
+    for own _id, user of @props.users
+      options.push @mapUserProperties(user)
 
+    selected = if @props.selectedUser?
+                 @mapUserProperties(@props.selectedUser)
+               else
+                 id: null, text: osu.trans('beatmap_discussions.user_filter.label')
```

Comment:
id optional? (also for `allUsers` object)

Reply:
id must be explicitly `null` for `BeatmapDiscussions.Main.update`